### ISSUE TITLE
abs_integrate hangs after encountering log(0)

### ISSUE THREAD

**Issue by rtoy** [rtoy/maxima#3577]
Imported from SourceForge on 2024-07-08 09:30:37
Created by **[kcrisman](https://sourceforge.net/u/kcrisman/)** on 2014-12-08 14:52:23
Original: https://sourceforge.net/p/maxima/bugs/2853

---

~~~~

(%i1) load(abs_integrate);
(%o1) /Users/karl.crisman/Downloads/sage/local/share/maxima/5.34.1/share/contr\
ib/integration/abs_integrate.mac
(%i4) integrate(log(abs(sin(x))), x, 0, %pi/2);

log: encountered log(0).
~~~~
And then it just hangs, forever as far as I can tell.

This is downstream at http://trac.sagemath.org/ticket/17468

Wolfram Alpha says this not only exists (as an improper integral) but is nice, -pi/2 log(2).

http://www.wolframalpha.com/input/?i=integrate+log%28abs%28sin%28x%29%29%29+from+0+to+pi%2F2

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-08 09:30:38
Created by **[willisbl](https://sourceforge.net/u/willisbl/)** on 2014-12-11 13:27:06
Original: https://sourceforge.net/p/maxima/bugs/2853/#bde0

---

~~~~
I'm testing a fix. In addition to the fix, I noticed that loading basic isn't needed anymore (moved push and pop to in core), and that the calls to supcontext need to be updated as well. After these changes, abs_integrate might not work correctly with older Maxima versions. That might cause trouble for Sage.

Incidentaly, without abs_integrate, Maxima likely  does integrate(log(abs(sin(x))),x,0,%pi/2) OK, but I'm not 100% sure of the value.

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-08 09:30:42
Created by **[kcrisman](https://sourceforge.net/u/kcrisman/)** on 2014-12-11 15:02:10
Original: https://sourceforge.net/p/maxima/bugs/2853/#49da

---

Hopefully won't cause trouble for Sage, as once the fix is in we would just upgrade everything together.  There is only one version of Maxima for any given version of Sage.

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-08 09:30:45
Created by **[kcrisman](https://sourceforge.net/u/kcrisman/)** on 2014-12-11 15:05:10
Original: https://sourceforge.net/p/maxima/bugs/2853/#ffb3

---

And the value Maxima gives without `abs_integrate` is correct, by the way, though perhaps it should 'know' some basic polylogarithm identities.

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-08 09:30:49
Created by **[willisbl](https://sourceforge.net/u/willisbl/)** on 2014-12-14 19:20:08
Original: https://sourceforge.net/p/maxima/bugs/2853/#9dc7

---

- **status**: open --> closed

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-08 09:30:53
Created by **[willisbl](https://sourceforge.net/u/willisbl/)** on 2014-12-14 19:20:09
Original: https://sourceforge.net/p/maxima/bugs/2853/#5dde

---

Fixed by Commit:eb935150c3d2c7717601da5bce470cb7f95f15df

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-08 09:30:56
Created by **[kcrisman](https://sourceforge.net/u/kcrisman/)** on 2014-12-15 16:18:58
Original: https://sourceforge.net/p/maxima/bugs/2853/#baf8

---

See the related ticket https://sourceforge.net/p/maxima/bugs/2859/
